### ISSUE TITLE
WebDAV '403 Forbidden' using rclone (only)

### ISSUE THREAD

**Issue by acs-ferreira** [rclone/rclone#3868]
#### What is the problem you are having with rclone?

Server is using a WAF and checking for `Referer` or `Origin` headers. If they (one or another) do not exist, requests are denied.

> Sorry, your request has neither Referer nor Origin set and is using a non safe method. Thus, your request has to be blocked.: 403 Forbidden

Accessing trough a browser, Nextcloud clients, Cyberduck or Filezilla Pro works fine.

#### What is your rclone version (output from `rclone version`)

`1.50.2`

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

`Ubuntu 18.04.3`

####  Which cloud storage system are you using? (eg Google Drive)

Webdav (Nextcloud)

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone lsd nextcloud:`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
rclone size nextcloud: -vv
2020/01/11 13:55:12 DEBUG : rclone: Version "v1.50.2" starting with parameters ["rclone" "size" "nextcloud:" "-vv"]
2020/01/11 13:55:12 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2020/01/11 13:55:12 ERROR : : error listing: couldn't list files: Sorry, your request has neither Referer nor Origin set and is using a non safe method. Thus, your request has to be blocked.: 403 Forbidden
2020/01/11 13:55:12 Failed to size with 2 errors: last error was: couldn't list files: Sorry, your request has neither Referer nor Origin set and is using a non safe method. Thus, your request has to be blocked.: 403 Forbidden
```

**Comment by ncw** [comment]
It would be easy enough to add a Referrer header (say the root of the WebDAV).

It looks like the code might be something like this: https://gist.github.com/simonerni/3501b8de6320ac37398d08d9d2d08561

> Accessing trough a browser, Nextcloud clients, Cyberduck or Filezilla Pro works fine.

Do you happen to know what any of these does (other than the browser which will sent Referrer headers)? I don't know if you can get them to dump their headers?

Try this which adds ` Referer` header pointing to the WebDAV root.

https://beta.rclone.org/branch/v1.50.2-126-g45aed723-fix-3868-webdav-referrer-beta/ (uploaded in 15-30 mins)

**Comment by acs-ferreira** [comment]
Will try you patch this week, thanks :)

**Comment by ncw** [comment]
How did you get on with the patch?

**Comment by acs-ferreira** [comment]
Link points to 404.

**Comment by ncw** [comment]
> Link points to 404.

I don't know what happened there!

Try this

https://beta.rclone.org/branch/v1.50.2-176-g63ad4ad3-fix-3868-webdav-referrer-beta/ (uploaded in 15-30 mins)

**Comment by acs-ferreira** [comment]
Seems OK, thanks :)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.51
